src/tests/ipc_tests.rs

```diff
@@ -408,4 +411,213 @@ pub mod ipc_tests {
+        // and also performs clean env setup
+        let _thelock = setup::lock_and_init();
+
+        let cage1 = interface::cagetable_getref(1);
+
+        let mut pipefds = PipeArray {
+            readfd: -1,
+            writefd: -1,
+        };
+        assert_eq!(cage1.pipe2_syscall(&mut pipefds, O_NONBLOCK), 0);
+        assert_eq!(cage1.fork_syscall(3), 0);
+
+        // dup our pipe read end to stdout and close unused pipe ends
+        assert_eq!(cage1.close_syscall(pipefds.readfd), 0);
+        assert_eq!(cage1.dup2_syscall(pipefds.writefd, 1), 1);
+        assert_eq!(cage1.close_syscall(pipefds.writefd), 0);
+
+        // lets write one byte more than the last page boundary of the pipe
+        let writesize1: usize = 61441;
+        let mut buf1: Vec<u8> = vec!['A' as u8; writesize1];
+        assert_eq!(
+            cage1.write_syscall(1, buf1.as_mut_ptr(), writesize1),
+            writesize1 as i32
+        );
+
+        // now if we try to write anything we expect EAGAIN because there isnt a page of
+        // room
+        let writesize2: usize = 4; // 4 bytes
+        let mut buf2: Vec<u8> = vec!['B' as u8; writesize2];
+        assert_eq!(
+            cage1.write_syscall(1, buf2.as_mut_ptr(), writesize2),
+            -(Errno::EAGAIN as i32)
+        );
+
+        assert_eq!(cage1.close_syscall(1), 0);
+
+        let sender = std::thread::spawn(move || {
+            let cage2 = interface::cagetable_getref(3);
+
+            // dup our pipe read end to stdout and close unused pipe ends
+            assert_eq!(cage2.close_syscall(pipefds.writefd), 0);
+            assert_eq!(cage2.dup2_syscall(pipefds.readfd, 0), 0);
+            assert_eq!(cage2.close_syscall(pipefds.readfd), 0);
+
+            let bytes_to_read = 65536;
+            let mut buf: Vec<u8> = Vec::with_capacity(bytes_to_read);
+
+            // ok now lets read everything
+            assert_eq!(
+                cage2.read_syscall(0, buf.as_mut_ptr(), bytes_to_read),
+                writesize1 as i32
+            );
+            // we've close the write ends so now we expect 0, signifying EOF
+            assert_eq!(
+                cage2.read_syscall(0, buf.as_mut_ptr(), bytes_to_read),
+                0 as i32
+            );
+
+            assert_eq!(cage2.close_syscall(0), 0);
+
+            assert_eq!(cage2.exit_syscall(EXIT_SUCCESS), EXIT_SUCCESS);
+        });
+
+        sender.join().unwrap();
+
+        assert_eq!(cage1.exit_syscall(EXIT_SUCCESS), EXIT_SUCCESS);
+
+        lindrustfinalize();
+    }
+
+    #[test]
+    pub fn ut_lind_ipc_pipe_rw_zero() {
+        //acquiring a lock on TESTMUTEX prevents other tests from running concurrently,
+        // and also performs clean env setup
+        let _thelock = setup::lock_and_init();
+
+        let cage1 = interface::cagetable_getref(1);
+
+        // lets setup the pipe and fork
+        let mut pipefds = PipeArray {
```

Comment:
I'm a bit confused about what this is doing.  In general, I'm against over-commenting tests, but why are readfd and writefd -1?)

Reply:
yeah i see that its unclear, just initializing here, added a comment